var/spack/repos/builtin/packages/py-google-auth/package.py

```diff
@@ -13,11 +13,18 @@ class PyGoogleAuth(PythonPackage):
     homepage = "https://github.com/GoogleCloudPlatform/google-auth-library-python"
     pypi = "google-auth/google-auth-1.6.3.tar.gz"
 
+    version('2.3.2', sha256='2dc5218ee1192f9d67147cece18f47a929a9ef746cb69c50ab5ff5cfc983647b')
     version('1.6.3', sha256='0f7c6a64927d34c1a474da92cfc59e552a5d3b940d3266606c6a28b72888b9e4')
 
     depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
+    depends_on('python@3.6:', type=('build', 'run'), when='@2.3.2:')
```

Comment:
`python_requires` still says 2.7 is supported:
```suggestion
    depends_on('python@2.7:2.8,3.6:', type=('build', 'run'), when='@2.3.2:')
```

Reply:
But the README says otherwise: https://github.com/googleapis/google-auth-library-python/blob/main/README.rst#supported-python-versions 

Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Python >= 3.6

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
- Python == 2.7:  The last version of this library with support for Python 2.7
  was `google.auth == 1.34.0`.

- Python 3.5:   The last version of this library with support for Python 3.5
  was `google.auth == 1.23.0`.

Reply:
Hmm, someone should tell the developers to update the important files, the README isn't checked by pip lol.

In that case I think you want:
```suggestion
    depends_on('python@3.6:', type=('build', 'run'), when='@1.35:')
    depends_on('python@2.7:2.8,3.6:', type=('build', 'run'), when='@1.24:')
```
Not sure when Python 3.4 support was dropped without digging through the history.